micro_http/src/request.rs

```diff
@@ -92,9 +98,16 @@ impl<'a> RequestLine<'a> {
     }
 
     fn parse_request_line(request_line: &[u8]) -> (&[u8], &[u8], &[u8]) {
-        let (method, remaining_bytes) = split(request_line, SP);
-        let (uri, remaining_bytes) = split(remaining_bytes, SP);
-        let (mut version, _) = split(remaining_bytes, LF);
+        let method_end = find(request_line, &[SP]);
+        let method = &request_line[..method_end];
+
+        let uri_and_version = &request_line[(method_end + 1)..];
+
+        let uri_end = find(uri_and_version, &[SP]);
+        let uri = &uri_and_version[..uri_end];
+
+        let version_end = find(uri_and_version, &[LF]);
+        let mut version = &uri_and_version[(uri_end + 1)..version_end];
```

Comment:
With these changes is the `remove_trailing_cr` method still needed?

Reply:
It is still used once in parsing the request line. However, changes can be made in order to stop using it, the most obvious one would be to pass a slice without the CRLF sequence to the request line parser.

Reply:
Yes, that's what I was thinking about. Please add an issue to refactor this code as a next step if you don't plan to do it as part of the same PR.